JavaScript/es6Guidelines.md

```diff
@@ -0,0 +1,189 @@
+
+## General Guidelines
+
+### `CONST` and `LET` over `VAR`
+
+Recommended article about var/let/const - [erric-elliot/javascript-es6-var-let-or-const](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75#.uw6acfhkw)
+
+We recommend following rule: `const > let > var`.  
+We listed some useful tips to help you decide which declaration to use.
+
+1.  **Are you going to reassign the variable's value?**  
+    **no)** use const  
+    **yes)** use let
+
+    _"No reassign" means - you are not going to change the reference._  
+
+    **Something like that is correct:**
+
+    ```javascript
+    const point = {
+      x: 5,
+      y: 5
+    };
+    point.x = 10
+    ```
+
+   **However, something like this is not:**
+
+    ```javascript
+    const shape = 'triangle';
+    shape = 'circle';
+    ```
+
+2.  **Always use one let/const/var declaration in block**  
+    You should put lets and consts on top of the block.  

```

Comment:
Are we sure about this rule? Because when converting to ES6 we didn't follow it at all. We have multiple places where we use it inline, like `for (let i=0; ...)` and we don't have any ticket to change it.